src/CONTRIBUTING.md

```diff
@@ -0,0 +1,154 @@
+
+#### Using `run:pause`
+
+When running commands, generating new files, or making changes to preexisting ones, you may want to see the current state of the generated app. The `run: pause` command is useful in this case. Use this command inside of your existing chapter, inserting it at points where you want the generator to stop. This is similar to running something like `pauseTest()`; it will pause the generator and allow you to see the current state of the code you have generated (located in `/dist/code/super-rentals`).
+
+For example:
+
+    ```run:pause
+    Note to self: check that `application.hbs` has the correct markup and remember to git add it!
+    ```
+
+Any content inside of the code blocks will be printed in your command line.
+
+#### Generating diffs
+
+In the case of generating diffs to render inside of the tutorial prose, you will want to use the `run: pause` command, described above.
+
+_⚠️ Please note: you may find it easier to generate diffs and make edits to the tutorial by opening `/dist/code/super-rentals/code` in a separate code editor window._
+
+Once you have put a `run: pause` at the spot in your prose where the diff needs to be captured, you can navigate to `/dist/code/super-rentals/code` in the terminal. First, check that the diff is included in the current changes:
+
+```
+➜  code git:(your-branch-name) ✗ git diff
+```
+
+Next, save your diff to some temporary file that you will open in a code editor in a moment (`my_awesome_patch` in the example below). You want to be sure to generate the diff specifically with only 1 line of context. You can do this by specifying `git diff --unified=1` or `git diff -U1`.
+
+```
+➜  code git:(your-branch-name) ✗ git diff -U1 > my_awesome_patch
+➜  code git:(your-branch-name) ✗ code my_awesome_patch
+```
+
+Once you have opened your temporary file containing your diff, edit it so that the only thing surrounding the diff is the line number additions subtractions.
+
+    ```
+    @@ -9,2 +9,3 @@
+    Router.map(function() {
+    +  this.route('about');
+    });
+    ```
+
+Finally, wrap the newly-modified diff content in the `run:file:patch` command. Be sure to specify the language of the file, the filename, and the `cwd=super-rentals` option so that the patch is applied in the app's directory.
+
+For example:
+
+    ```run:file:patch lang=js cwd=super-rentals filename=app/router.js
+    @@ -9,2 +9,3 @@
+    Router.map(function() {
+    +  this.route('about');
+    });
+    ```
+
+Once you have added the diff to the prose using the `run:file:patch` command, you can remove the `run:pause` command. In order to test that your patch was successfully applied, be sure to:
+
+1. Run `git checkout` any changes in the workoing directory of `/dist/code/super-rentals/code`.
+2. Regenerate your modified chapter (`yarn generate src/chapters/00-your-chapter-to-edit-here.md`).
```

Comment:
This may want to be close to the generation workflow? (Rest that it works before editing/pasting into the code block)